src/Model/Ticket.java

```diff
@@ -97,88 +86,69 @@ public void setStadiumSector(StadiumSector stadiumSector) {
         this.stadiumSector = stadiumSector;
     }
 
- /*   public void setUnnamedStadiumSector() {
-        this.stadiumSector = '-';
-    }
-*/
     public double getbackpackAllowedWeight() {
         return backpackAllowedWeight;
     }
-
-    public void setbackpackAllowedWeight(double backpackAllowedWeight) {
-        if (backpackAllowedWeight < 0) {
-            System.out.println("Invalid backpack weight.");
-            System.exit(0);
-        }
-        this.backpackAllowedWeight = backpackAllowedWeight;
-    }
-
+    
     public String getPrice() {
         return FORMATTER.format(this.price);
     }
 
-    public void setPrice(Integer price) {
-        if (price < 0 )
-        {
-            System.out.println("Invalid price.");
-            System.exit(0);
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == null) {
+            return false;
+        }
+
+        if (!(object instanceof Ticket)) {
+            return false;
+        }
+
+        if (this == object) {
+            return true;
+        }
+
+        if (concertHall == ((Ticket) object).concertHall &&
+                eventCode == ((Ticket) object).eventCode &&
+                eventTime.equals(((Ticket) object).eventTime) &&
+                isPromo == ((Ticket) object).isPromo &&
+                stadiumSector == ((Ticket) object).stadiumSector &&
+                backpackAllowedWeight == ((Ticket) object).backpackAllowedWeight &&
+               price.equals(((Ticket) object).price)) {
+            return true;
         }
         else {
-            this.price = new BigDecimal(price);
+            return false;
         }
     }
 
-    public Ticket() {
-        this(ConcertHall.not_specified, 3, null, false, StadiumSector.not_specified, 0); // How to create a Ticket without Event Time, not using "null"??
+  @Override
+    public int hashCode() {
+        return getId().hashCode();
     }
 
-    public Ticket(ConcertHall concertHall, int eventCode, LocalDateTime eventTime) {
-        this(concertHall, eventCode, eventTime, false, StadiumSector.not_specified, 0);
+    public void share(PhoneNumber phoneNumber) {
+        System.out.println("Ticket "  + getId() + " was shared by phone number");
     }
 
-    public Ticket(ConcertHall concertHall, int eventCode, LocalDateTime eventTime, boolean isPromo, StadiumSector stadiumSector, double backpackAllowedWeight) {
-        ticketId = ticketsCounter;
-        ticketsCounter++;
+    public void share(PhoneNumber phoneNumber, Email email) {
+        System.out.println("Ticket " + getId() + " was shared by phone number and email");
```

Comment:
if you declare arguments, you need to do smth with them inside the methos. For example, print them also in your message.

Reply:
Fixed. I updated overloaded methods share() to use all input parameters